base/deprecated.jl

```diff
@@ -1009,4 +1009,18 @@ export @vectorize_1arg, @vectorize_2arg
 @deprecate abs(M::SymTridiagonal) abs.(M)
 @deprecate abs(x::AbstractSparseVector) abs.(x)
 
+# #18218
+eval(Base.LinAlg, quote
+    function arithtype(T)
+        depwarn("arithtype is deprecated as it is no longer needed in Julia Base.",
```

Comment:
Instead of using `arithtype` which replaces `Bool` by `Int` for matrix multiplications, we use an internal method `matprod(x, y) = x*y + x*y` which should cover the roll of `arithtype` while also being more general.